app/ts/Subproblem/SubproblemButtons/AddSubproblemButton/AddSubproblemUtil.test.ts

```diff
@@ -0,0 +1,619 @@
+import IAlternative from '@shared/interface/IAlternative';
+import ICriterion from '@shared/interface/ICriterion';
+import IRelativePerformance from '@shared/interface/IRelativePerformance';
+import ISubproblemDefinition from '@shared/interface/ISubproblemDefinition';
+import IWorkspace from '@shared/interface/IWorkspace';
+import {
+  createSubproblemDefinition,
+  getBaselineMap,
+  getMissingValueWarnings,
+  getScaleBlockingWarnings,
+  initInclusions,
+  isAlternativeDisabled,
+  isDataSourceDeselectionDisabled
+} from './AddSubproblemUtil';
+
+describe('addSubproblemUtil', () => {
+  describe('getMissingValueWarnings', () => {
+    const noSmaaWarning =
+      'Some cell(s) are missing SMAA values. Deterministic values will be used for these cell(s).';
+    const noDeterministicWarning =
+      'Some cell(s) are missing deterministic values. SMAA values will be used for these cell(s).';
+
+    it('should return no warnings if all values are present', () => {
+      const dataSourceInclusions = {ds1Id: true, ds2Id: false};
+      const alternativeInclusions = {alt1Id: true};
+
+      const workspace = {
+        effects: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          },
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds2Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          }
+        ],
+        distributions: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          }
+        ],
+        relativePerformances: []
+      } as IWorkspace;
+
+      const warnings = getMissingValueWarnings(
+        dataSourceInclusions,
+        alternativeInclusions,
+        workspace
+      );
+
+      const expectedWarnings: string[] = [];
+      expect(warnings).toEqual(expectedWarnings);
+    });
+
+    it('should warn about missing SMAA values when Deterministic values are present', () => {
+      const dataSourceInclusions = {ds1Id: true};
+      const alternativeInclusions = {alt1Id: true};
+
+      const workspace = {
+        effects: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          }
+        ],
+        distributions: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'empty'
+          }
+        ],
+        relativePerformances: []
+      } as IWorkspace;
+
+      const warnings = getMissingValueWarnings(
+        dataSourceInclusions,
+        alternativeInclusions,
+        workspace
+      );
+
+      const expectedWarnings: string[] = [noSmaaWarning];
```

Comment:
in such cases I'd argue that the local variable actually makes things harder to read due to indirection

Reply:
Not sure I understand you here.